src/herbpy/tsr/block.py

```diff
@@ -1,7 +1,47 @@
 import numpy
 import prpy.tsr 
+from prpy.tsr.tsrlibrary import TSRFactory
 
-@prpy.tsr.tsrlibrary.TSRFactory('herb', 'block', 'place')
+@TSRFactory('herb', 'block', 'grasp')
+def block_grasp(robot, block, manip=None):
+    """
+    Generates end-effector poses for moving the arm near the block
+    @param robot The robot grasping the block
+    @param block The block being grasped
+    @param manip The manipulator to move near the block, if None
+       the active manipulator of the robot is used
+    """
+    if manip is None:
+        manip_idx = robot.GetActiveManipulatorIndex()
+        manip = robot.GetActiveManipulator()
+    else:
+        with manip.GetRobot():
+            manip.SetActive()
+            manip_idx = manip.GetRobot().GetActiveManipulatorIndex()
+
+    offset = 0.01 #vertical offset relative to block
+    alpha = 0.8 # orientation of end-effector relative to block
+
+    block_in_world = block.GetTransform()
+    block_in_world[:3,:3] = numpy.eye(3) # ignore orientation
+    ee_in_block = numpy.array([[numpy.cos(alpha), 0., -numpy.sin(alpha), 0.3*numpy.sin(alpha)+0.04],
+                               [ 0., -1, 0, 0.],
+                               [-numpy.sin(alpha), 0., -numpy.cos(alpha), 0.25+offset],
+                               [ 0., 0., 0., 1.]])
+    Bw = numpy.zeros((6,2))
+    Bw[5,:] = [-numpy.pi, numpy.pi-.0001]
```

Comment:
This `- 0.001` should not be required as of https://github.com/personalrobotics/prpy/pull/194.